### PR TITLE
Filled in all classes, implemented everything except attacking

### PR DESCRIPTION
**Pull request by Crollt**

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -5,44 +5,139 @@
...
+        for (Result res : this.attacks) {
+            if (res.getLocation() == attackLoc) {
+                result.setResult(AtackStatus.INVALID);
+                return result;
+            }
+        }
+        // Continue if attack is valid
+	    for (Ship ship : this.ships) {
+	        boolean hit = false;
+	        // Check if the attack hits a ship
```
Comment (Bizjakj) [inline src/main/java/cs361/battleships/models/Board.java]: will still need to implement code here

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -5,44 +5,139 @@
...
+            for (Square occSquare : ship.getOccupiedSquares()) {
+                if (occSquare == attackLoc) {
+                    result.setShip(ship);
+                    hit = true;
+                    break;
+                }
+            }
+            if (hit) {
+                // Ship has been hit, check if sunk
```
Comment (Bizjakj) [inline src/main/java/cs361/battleships/models/Board.java]: also need to check if all ships have been sunk here to see if we need to end the game

File: src/main/java/cs361/battleships/models/Board.java (hunk 1 above)
Comment (Crollt) [reply src/main/java/cs361/battleships/models/Board.java]: Yep. That's why my commit message says implemented everything except attack.

File: src/main/java/cs361/battleships/models/Board.java (hunk 2 above)
Comment (Crollt) [reply src/main/java/cs361/battleships/models/Board.java]: Feel free.

Comment (Bizjakj) [reply src/main/java/cs361/battleships/models/Board.java]: ok, will do

**Pull request merged**